test/services/charge-elements/add-licences-to-workflow.service.test.js

```diff
@@ -0,0 +1,97 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+const Sinon = require('sinon')
+
+const { describe, it, beforeEach, afterEach } = exports.lab = Lab.script()
+const { expect } = Code
+
+// Test helpers
+const DatabaseHelper = require('../../support/helpers/database.helper.js')
+const { generateUUID } = require('../../../app/lib/general.lib.js')
+const Workflow = require('../../../app/models/water/workflow.model.js')
+
+// Things we need to stub
+const FetchLicencesForWorkflowService = require('../../../app/services/charge-elements/fetch-licences-for-workflow.service.js')
+
+// Thing under test
+const AddLicencesToWorkflowService = require('../../../app/services/charge-elements/add-licences-to-workflow.service.js')
+
+describe('Add Licences to Workflow service', () => {
+  let notifierStub
+
+  beforeEach(() => {
+    // The service depends on GlobalNotifier to have been set. This happens in app/plugins/global-notifier.plugin.js
+    // when the app starts up and the plugin is registered. As we're not creating an instance of Hapi server in this
+    // test we recreate the condition by setting it directly with our own stub
+    notifierStub = { omfg: Sinon.stub() }
+    global.GlobalNotifier = notifierStub
+  })
+
+  afterEach(async () => {
+    await DatabaseHelper.clean()
```

Comment:
`await DatabaseHelper.clean()` needs to go in the `beforeEach()` not the `afterEach()`.

That way

- the test is always starting with a clean slate
- should the test fail the data remains for possible debugging purposes

Reply:
I think I was lazy and used another test as a template, so I suspect this issue also exists in other tests. I'll have a look and fix anything I find.